src/Daemon/DockerRunJob.ts

```diff
@@ -0,0 +1,32 @@
+import Dockerode from 'dockerode'
+import { ServerChannel } from 'ssh2'
+
+import { attachStreams, createContainer } from './DaemonExec'
+import { RunJob } from './RunJob'
+import { JobRequest } from '../Network'
+
+function argvForMake(target: string): string[] {
+	return ['make', target]
+}
+
+/**
+ * Factory for making a job runner using Docker.
+ * The daemon would use this to know how to run a job.
+ * @param docker - A connected Dockerode client.
+ * @returns A job runner that uses Docker.
+ */
+export function dockerRunJob(docker: Dockerode): RunJob {
+	return async (request: JobRequest, channel: ServerChannel) => {
+		await docker.pull(request.image)
```

Comment:
Interesting, I used `ensureImageImport` at first but then temporarily switched to this, because of the same issue you're mentioning. I'll do some more careful testing of this.

Reply:
Yeah you're totally right, `docker.pull` doesn't resolve reliably. Fixed this.